core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -253,84 +238,61 @@ public void abort(Throwable cause) {
             throw new UnsupportedOperationException();
         }
 
-        /**
-         * Writes the specified {@link HttpObject} to {@link DecodedHttpResponse}. This method is only called
-         * from {@link Http1ResponseDecoder} and {@link Http2ResponseDecoder}. If this returns {@code false},
-         * it means the response stream has been closed due to disconnection or by the response consumer.
-         * So the caller do not need to handle such cases because it will be notified to the response
-         * consumer anyway.
-         */
         @Override
         public boolean tryWrite(HttpObject o) {
-            boolean wrote = false;
-            switch (state) {
-                case WAIT_NON_INFORMATIONAL:
-                    wrote = handleWaitNonInformational(o);
-                    break;
-                case WAIT_DATA_OR_TRAILERS:
-                    wrote = handleWaitDataOrTrailers(o);
-                    break;
-                case DONE:
-                    PooledObjects.close(o);
-                    break;
+            if (done) {
+                PooledObjects.close(o);
+                return false;
             }
-
-            return wrote;
-        }
-
-        @Override
-        public boolean tryWrite(Supplier<? extends HttpObject> o) {
             return delegate.tryWrite(o);
         }
 
-        private boolean handleWaitNonInformational(HttpObject o) {
-            // NB: It's safe to call logBuilder.startResponse() multiple times.
+        void startResponse() {
+            if (responseStarted) {
+                return;
+            }
+            responseStarted = true;
             if (ctx != null) {
                 ctx.logBuilder().startResponse();
+                ctx.logBuilder().responseFirstBytesTransferred();
             }
+            initTimeout();
+        }
+
+        boolean tryWriteResponseHeaders(ResponseHeaders responseHeaders) {
```

Comment:
Could you just drop `try` prefix since we don't have `writeResponseHeaders`?

Reply:
No, we don't have it but I wanted to align with `AbstractStreamWriter.tryWrite(..)`.
Please let me know if you are strong on this.

Reply:
No, it's fine as it is now.